plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -272,16 +287,52 @@ public function register_checkout_field( $options ) {
 			trigger_error( sprintf( 'Registering a field with hidden set to true is not supported. The field "%s" will be registered as visible.', esc_html( $id ) ), E_USER_WARNING );
 		}
 
-		// Insert new field into the correct location array.
-		$this->additional_fields[ $id ] = array(
+		$field_data = array(
 			'label'          => $options['label'],
 			'hidden'         => false,
+			'type'           => $type,
 			'optionalLabel'  => empty( $options['optionalLabel'] ) ? '' : $options['optionalLabel'],
 			'required'       => empty( $options['required'] ) ? false : $options['required'],
 			'autocomplete'   => empty( $options['autocomplete'] ) ? '' : $options['autocomplete'],
 			'autocapitalize' => empty( $options['autocapitalize'] ) ? '' : $options['autocapitalize'],
 		);
 
+		if ( 'select' === $type ) {
+			if ( empty( $options['options'] ) || ! is_array( $options['options'] ) ) {
+				return new \WP_Error( 'woocommerce_blocks_checkout_select_field_no_options_specified', __( 'Fields of type "select" must have an array of "options".', 'woocommerce' ) );
```

Comment:
What are your thoughts on adding some line breaks for readability-reasons?

```suggestion
				return new \WP_Error(
					'woocommerce_blocks_checkout_select_field_no_options_specified',
					__( 'Fields of type "select" must have an array of "options".', 'woocommerce' )
				);
```